Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/dao/ProductServiceRepository.java

```diff
@@ -0,0 +1,15 @@
+package com.shopkeeper.productservice.dao;
+
+import java.util.List;
+
+import org.springframework.data.repository.PagingAndSortingRepository;
+import org.springframework.stereotype.Repository;
+
+import com.shopkeeper.productservice.dto.Product;
+
+@Repository
+public interface ProductServiceRepository extends PagingAndSortingRepository<Product, Integer>{
```

Comment:
ProductServiceRepository or ProductRepository?

Reply:
Why are we using PagingAndSortingRepository here? Will a JPARepository not be enough? What are the differences?

Reply:
The `PagingAndSortingRepository` extend the `CrudRepository`. So, along with providing the additional methods to sort and page the result, it also provides the basic CRUD operations. `JPARepository` further implements the `PagingAndSortingRepository`, so along with getting all the functions provided by it, we also get additional functions that come with the `JPARepository` like performing delete operations in batch, etc. We don't need that according to our use-case.